### PR TITLE
aes_string changes

### PR DESCRIPTION
**Pull request by kenzeng24**
replaced all instances of aes_string(...) with aes(...).

### CODE REVIEW

**Comment by michaelweylandt** [comment]
Hi Ken, 

This looks like a good start, but a few things to check out: 

1) Can you edit this PR so that the target branch is `develop` instead of `master`? I try to reserve the master branch for "stable" versions while `develop` is used for unreleased work. (The master branch also has autogenerated files.)

2) The `NAMESPACE` file needs to import `aes` instead of `aes_string` now. You can change this by changing the `roxygen2` imports from `@importFrom ggplot2 aes_string` to `@importFrom ggplot2 aes` everywhere that incantation appears. 

3) Does this pass `R CMD check`? (Changing the branch target to `develop` will set things up to automatically run checks for you.) If I remember correctly, we first switched to `aes_string` in order to avoid a unused global variable warning. The plain `aes` calls seem like they would get the same warning. To avoid the warning, I think you'll need to look into the "quasiquotation" / "tidy evaluation" ideas described in the `?aes` help page.

**Comment by michaelweylandt** [comment]
Linking to #93

**Comment by michaelweylandt** [comment]
Discussion relevant to point 3 above: https://stackoverflow.com/questions/9439256/how-can-i-handle-r-cmd-check-no-visible-binding-for-global-variable-notes-when

The penultimate answer (https://stackoverflow.com/a/57496617/967712) seems like it would work for us (with some modifications) and be pretty straightforward.

**Comment by kenzeng24** [comment]
Thank you for your suggestions. Here are some of the issues I came across: 
After including the code for global variable binding, running devtools::check() gives me one note and one error: 

	error: No matching test file in dir
	note: Packages suggested but not available for checking:  'cvxclustr', ‘cvxbiclustr'

Have you encountered these before? 

Regards

Ken

**Comment by michaelweylandt** [comment]
That sounds like two distinct problems

- The first part is probably because the test running file (https://github.com/DataSlingers/clustRviz/blob/develop/tests/test_stub.R) is a "stub" that gets transformed to run all tests during CI. For now, easiest thing to do is probably just remove the `filter="PATTERN"` argument locally if you want to run all tests. 

- The second thing just means you need to install those two packages to run all the tests. 

Both lines of logic are spelled out in the `.github/workflows` files

File: R/plot_carp.R [hunk 1]
```diff
@@ -80,6 +80,11 @@
 #' carp_fit <- CARP(presidential_speech)
 #' plot(carp_fit, type='interactive')
 #' }
+
+if(getRversion() >= "2.15.1")  {
+  utils::globalVariables(c("V1", "V2", "Obs", "FirstV1", "FirstV2", "final_cluster", "ObsLabel", "FirstObsLabel"))
+}
+
```
Comment (michaelweylandt) [inline R/plot_carp.R:87]: This is a pretty ugly hack. Did you look at the fix I recommended? (https://stackoverflow.com/questions/9439256/how-can-i-handle-r-cmd-check-no-visible-binding-for-global-variable-notes-when/57496617#57496617)?

**Comment by kenzeng24** [comment]
I tried the line: 

	if(getRversion() >= "2.15.1") utils::globalVariables(c("."))

Before this ugly hack. Is this the line you were referring to? 

Regards
Ken

**Comment by michaelweylandt** [comment]
I'm saying using `globalVariables` in general is a really ugly hack and something we shouldn't do. When you click the link I've posted twice, does it not go to the answer about `.data`? That's what I've been trying to suggest / push you towards.

**Comment by michaelweylandt** [comment]
Quick look just now - the CI builds look solid. The Windows failure is a very small floating point thing (error of 1e-16) while the Mac devel test looks like a lack of CRAN binaries, not a true failure.

**Comment by kenzeng24** [comment]
Thank you for the suggestions. 
Do you know how to create a new branch that’s a copy of develop? Currently when I try to create a new branch it always ends up being a copy of main. 

Regards
Ken

**Comment by michaelweylandt** [comment]
It depends a bit on which interface you are using. 

From the command line, first checkout develop

    git checkout develop

Then create and checkout a new branch

   git checkout -b kz/fix_aes_string

Then start adding commits to that branch. You can "move" commits from one branch to another, but it's subtle and the changes you actually want to send are small, so I'd just re-do them.

If you're using RStudio, the steps are similar. First switch to the "develop" branch in the git panel, then hit "New Branch". 

Both of these methods assume you have a develop branch to start from. If you cloned the repo the normal way (as opposed to just downloading all the files from the GitHub website directly) git should know where to find `develop` automatically for you.

**Comment by michaelweylandt** [comment]
Closed in lieu of #105.

**Pull request closed without merging**